inc/blokus.h

```diff
@@ -0,0 +1,83 @@
+#ifndef _BLOKUS_H
+#define _BLOKUS_H 
+
+#define N_ROW 14
+#define N_COL 14
+
+#define within_map(y, x) ((y) >= 0 && (y) < N_ROW && (x) >= 0 && (x) < N_COL)
+
+typedef enum {
+    // monomino
+    SHAPE_M,
+    // domino
+    SHAPE_D,
+    // trominoes
+    SHAPE_R,
+    SHAPE_A,
+    // tetrominoes
+    SHAPE_E,
+    SHAPE_J,
+    SHAPE_S,
+    SHAPE_Q,
+    SHAPE_H,
+    // pentominoes
+    SHAPE_F,
+    SHAPE_I,
+    SHAPE_L,
+    SHAPE_N,
+    SHAPE_P,
+    SHAPE_T,
+    SHAPE_U,
+    SHAPE_V,
+    SHAPE_W,
+    SHAPE_X,
+    SHAPE_Y,
+    SHAPE_Z,
+} shape_t;
+
+typedef struct {
+    int y;
+    int x;
+} coord_t;
+
+struct _tile_attr {
+    char alpha;
+    coord_t blks[5];
+    int blk_cnt;
+    int can_mir;
+    int rot_cnt;
+};
+
+typedef struct {
+    shape_t shape;
+    coord_t *blks;
+    coord_t pos;
+} tile_t;
+
+typedef struct {
+    int turn;
+    shape_t sel_shape;
+    tile_t *hand[2][SHAPE_Z + 1];
+    int score[2];
+    int next_empty[N_ROW * N_COL];
+    int prev_empty[N_ROW * N_COL];
+    int map[N_ROW][N_COL];
+    char code[16];
```

Comment:
We may add error code attribute in `gcb` so that I can give user corresponding messages

Reply:
I think I will include this in another feature PR. Do you think this is OK?

Reply:
That'll be great, thanks